templates/flavors/kubeadm/konnectivity/kustomization.yaml

```diff
@@ -0,0 +1,119 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - ../default
+  - ../../../addons/konnectivity
+
+patches:
+  - target:
+      group: controlplane.cluster.x-k8s.io
+      version: v1beta1
+      kind: KubeadmControlPlane
+    patch: |-
+      apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+      kind: KubeadmControlPlane
+      metadata:
+        name: ${CLUSTER_NAME}-control-plane
+      spec:
+        kubeadmConfigSpec:
+          preKubeadmCommands:
+            - /kubeadm-pre-init.sh ${KUBERNETES_VERSION}
+            - sed -i '/swap/d' /etc/fstab
+            - swapoff -a
+            - hostnamectl set-hostname '{{ ds.meta_data.label }}' && hostname -F /etc/hostname
+            - mkdir -p -m 755 /etc/kubernetes/konnectivity
+            - curl -s -L https://raw.githubusercontent.com/linode/konnectivity/876b00f976975833929723ad9e247d0c124124bd/config/egress-selector-configuration.yaml > /etc/kubernetes/konnectivity/egress-selector-configuration.yaml
```

Comment:
I know we already merged something like this for the kubeadm base flavor, but why do we want to lock this to a specific commit version instead of `main` or a tag?

Reply:
We want to make sure updates to these scripts don't break things for someone using an older release. So we are using git commit hash which is helping us to pin to a specific tested version (same what a tag would do).

Reply:
out of scope for this PR, but I think we should produce this is a release artifact and eventually reference that directly instead of using a raw commit file